src/main/java/org/broadinstitute/consent/http/service/DarCollectionService.java

```
@@ -215,6 +217,61 @@ private List<Integer> getDacIdsFromUser (User user) {
             .collect(Collectors.toList());
   }
 
+  public void deleteByCollectionId(User user, Integer collectionId) throws NotAcceptableException, NotAuthorizedException {
+    // ensure the user is capable of deleting the collection
+    if (!user.hasUserRole(UserRoles.ADMIN)) {
+      DarCollection coll = darCollectionDAO.findDARCollectionByCollectionId(collectionId);
```

Move this up and fail fast with a `NotFoundException`